### PR TITLE
updates for hf module

### PR DESCRIPTION
**Pull request by gorkem**
Updates the dagger engine and dependencies. Improves error handling.

### CODE REVIEW

File: huggingface/dagger/go.mod [hunk 1]
```diff
@@ -17,24 +19,35 @@ require (
...
+	google.golang.org/protobuf v1.34.2 // indirect
 )
+
+replace go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc => go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc v0.0.0-20240518090000-14441aefdf88
+
+replace go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp => go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploghttp v0.3.0
+
+replace go.opentelemetry.io/otel/log => go.opentelemetry.io/otel/log v0.3.0
+
+replace go.opentelemetry.io/otel/sdk/log => go.opentelemetry.io/otel/sdk/log v0.3.0
```
Comment (amisevsk) [inline huggingface/dagger/go.mod:53]: These can be put into one `replace` block
```go.mod
replace (
  // replacements
)
```

File: huggingface/dagger/go.mod [hunk 2]
```diff
@@ -1,11 +1,13 @@
 module dagger/huggingface
 
-go 1.22.4
+go 1.22.5
+
+toolchain go1.23.2
```
Comment (amisevsk) [inline huggingface/dagger/go.mod:5]: Should these versions be the same?

Comment (gorkem) [reply huggingface/dagger/go.mod:5]: updated by `dagger develop` command

File: huggingface/dagger/go.mod (hunk 1 above)
Comment (gorkem) [reply huggingface/dagger/go.mod:53]: We can but `dagger develop` will restore it next time we invoke it

Comment (amisevsk) [reply huggingface/dagger/go.mod:53]: Perfect

**Review by amisevsk — APPROVED** [review]

**Pull request merged**